internal/services/containerapps/container_app_environment_managed_certificate_resource.go

```diff
@@ -0,0 +1,231 @@
+		},
+
+		"domain_control_validation_type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringInSlice(managedenvironments.PossibleValuesForManagedCertificateDomainControlValidation(), false),
+			Description:  "Type of domain control validation. Possible values include `CNAME`, `HTTP` and `TXT`.",
+		},
+
+		"subject_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "Subject name of the certificate.",
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (a ContainerAppEnvironmentManagedCertificateResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (a ContainerAppEnvironmentManagedCertificateResource) ModelObject() interface{} {
+	return &ContainerAppEnvironmentManagedCertificateModel{}
+}
+
+func (a ContainerAppEnvironmentManagedCertificateResource) ResourceType() string {
+	return "azurerm_container_app_environment_managed_certificate"
+}
+
+func (a ContainerAppEnvironmentManagedCertificateResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return managedenvironments.ValidateManagedCertificateID
+}
+
+func (a ContainerAppEnvironmentManagedCertificateResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			managedEnvironmentsClient := metadata.Client.ContainerApps.ManagedEnvironmentClient
+
+			var cert ContainerAppEnvironmentManagedCertificateModel
+
+			if err := metadata.Decode(&cert); err != nil {
+				return err
+			}
+
+			managedEnvironmentId, err := managedenvironments.ParseManagedEnvironmentID(cert.ManagedEnvironmentId)
+			if err != nil {
+				return err
+			}
+
+			managedEnvironment, err := managedEnvironmentsClient.Get(ctx, *managedEnvironmentId)
+			if err != nil {
+				return fmt.Errorf("reading %s: %+v", *managedEnvironmentId, err)
+			}
+
+			managedCertId := managedenvironments.NewManagedCertificateID(metadata.Client.Account.SubscriptionId, managedEnvironmentId.ResourceGroupName, managedEnvironmentId.ManagedEnvironmentName, cert.Name)
+
+			existingManagedCert, err := managedEnvironmentsClient.ManagedCertificatesGet(ctx, managedCertId)
+			if err != nil {
+				if !response.WasNotFound(existingManagedCert.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", managedCertId, err)
+				}
+			}
+			if !response.WasNotFound(existingManagedCert.HttpResponse) {
+				return metadata.ResourceRequiresImport(a.ResourceType(), managedCertId)
+			}
+
+			domainControlValidation := managedenvironments.ManagedCertificateDomainControlValidation(cert.DomainControlValidation)
+
+			model := managedenvironments.ManagedCertificate{
+				Location: managedEnvironment.Model.Location,
```

Comment:
We need to normalise the location here
```suggestion
				Location: location.Normalize(managedEnvironment.Model.Location),
```